pkg/resource/statefulset_test.go

```diff
@@ -38,6 +39,8 @@ import (
 var update = flag.Bool("update", false, "update the golden files of this test")
 
 func TestStatefulSetBuilder(t *testing.T) {
+	// turn on featuregate to test rules
+	utilfeature.DefaultMutableFeatureGate.Set("AffinityRules=true")
```

Comment:
Is there a test where we have the affinity rules set in a cluster spec but the feature flag is disabled and things work as expected? i.e. the affinity rules are not passed to the pod

Reply:
We can test that the feature is disabled, but again we have test for the featuregates already